pkg/pdp/service/provider_register.go

```diff
@@ -0,0 +1,161 @@
+			Description: providerInfoView.Info.Description,
+		}, nil
+	}
+
+	// not registered, lets do this
+	abiData, err := bindings.ServiceProviderRegistryMetaData.GetAbi()
+	if err != nil {
+		return types.RegisterProviderResults{}, fmt.Errorf("failed to get ABI: %w", err)
+	}
+
+	/*
+	 /// @notice Register as a new service provider with a specific product type
+	    /// @param payee Address that will receive payments (cannot be changed after registration)
+	    /// @param name Provider name (optional, max 128 chars)
+	    /// @param description Provider description (max 256 chars)
+	    /// @param productType The type of product to register
+	    /// @param productData The encoded product configuration data
+	    /// @param capabilityKeys Array of capability keys
+	    /// @param capabilityValues Array of capability values
+	    /// @return providerId The unique ID assigned to the provider
+	    function registerProvider(
+	        address payee,
+	        string calldata name,
+	        string calldata description,
+	        ProductType productType,
+	        bytes calldata productData,
+	        string[] calldata capabilityKeys,
+	        string[] calldata capabilityValues
+	    ) external payable returns (uint256 providerId) {
+	*/
+
+	/*
+		The PDPOffering data (serviceURL, minPieceSizeInBytes,
+		  maxPieceSizeInBytes, storagePricePerTibPerMonth, etc.) is completely
+		  ignored.
+		  Instead, FilecoinWarmStorageService uses its own:
+		  - Fixed pricing: 5 USDFC per TiB/month (line 302)
+		  - Fixed proving periods: Set during initialization (lines 345-346)
+		  - No piece size restrictions from PDPOffering
+		  So, the PDPOffering is just stored in the
+		  registry for discovery/informational purposes. It's not used
+		  operationally by the FilecoinWarmStorageService contract.
+		  This makes sense architecturally because:
+		  1. The registry acts as a "yellow pages" where providers advertise their
+		  capabilities
+		  2. The actual service contract (FilecoinWarmStorageService) enforces its
+		  own standardized terms
+		  3. Clients might use PDPOffering data to discover providers, but the
+		  actual service operates on fixed terms
+	*/
+	productData, err := registry.EncodePDPOffering(bindCtx, bindings.ServiceProviderRegistryStoragePDPOffering{
+		// so I don't think any of these fields matter, see above comment
+		// TODO validate this assumption, I don't think it's used, but need to verify
+		ServiceURL:                 "http://example.com",
+		MinPieceSizeInBytes:        big.NewInt(1),
+		MaxPieceSizeInBytes:        big.NewInt(2),
+		IpniPiece:                  false,
+		IpniIpfs:                   false,
+		StoragePricePerTibPerMonth: big.NewInt(2),
+		MinProvingPeriodInEpochs:   big.NewInt(30),
+		Location:                   "earth",
+		PaymentTokenAddress:        p.address,
+	})
+	if err != nil {
+		return types.RegisterProviderResults{}, fmt.Errorf("failed to encode product data: %w", err)
+	}
```

Comment:
Based on the contract code, the only field that actually matters is the payee: https://github.com/storacha/filecoin-services/blob/main/service_contracts/src/FilecoinWarmStorageService.sol#L492, which isn't in this struct, but defined below on line 116 as `p.address` - the address of the operator running the node.

I don't know if we want to let providers populate this with their own preferences (that don't actually matter), or hard code it with (less sarcastic) values than I have used here.